### PR TITLE
add vulnerability certifier parser

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 <parth.psu@gmail.com>

Adds vulnerability certifier parser that generates the package node, attestation node, and vulnerability nodes based on the vulnerability attestation defined. The package node is merged into the existing node based on the purl.

Part of Issue #222

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/ingestor/parser/vuln/cerify_vuln.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	c.getAttestation(doc.Blob, doc.SourceInformation.Source, statement)
+	c.getVulns(doc.Blob, doc.SourceInformation.Source, statement)
+	return nil
+}
+
+func (c *vulnCertificationParser) getSubject(statement *attestation_vuln.VulnerabilityStatement) {
+	currentPackage := assembler.PackageNode{}
+	for _, sub := range statement.Subject {
+		currentPackage.Purl = sub.Name
+		if len(sub.Digest) > 0 {
```
Comment (lumjjb) [inline pkg/ingestor/parser/vuln/cerify_vuln.go:71]: nit : check not required. range of a nil or len 0 will not run

File: pkg/ingestor/parser/vuln/cerify_vuln.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	attNode.Payload["invocation_uri"] = statement.Predicate.Invocation.Uri
+	attNode.Payload["invocation_eventID"] = statement.Predicate.Invocation.EventID
+	attNode.Payload["invocation_producerID"] = statement.Predicate.Invocation.ProducerID
+	attNode.Payload["scanner_uri"] = statement.Predicate.Scanner.Uri
+	attNode.Payload["scanner_version"] = statement.Predicate.Scanner.Version
+	attNode.Payload["scanner_db_uri"] = statement.Predicate.Scanner.Database.Uri
+	attNode.Payload["scanner_db_version"] = statement.Predicate.Scanner.Database.Version
+	attNode.Payload["metadata_scannedOn"] = statement.Predicate.Metadata.ScannedOn.String()
+	for i, result := range statement.Predicate.Scanner.Result {
+		attNode.Payload["result_vulnerabilityID"+strconv.Itoa(i)] = result.VulnerabilityId
```
Comment (lumjjb) [inline pkg/ingestor/parser/vuln/cerify_vuln.go:99]: delimeter for readability?

File: pkg/ingestor/parser/vuln/cerify_vuln.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		attNode.Payload["result_alias"+strconv.Itoa(i)] = result.Aliases
+	}
+	c.attestation = attNode
+}
+
+func (c *vulnCertificationParser) getVulns(blob []byte, source string, statement *attestation_vuln.VulnerabilityStatement) {
+	for _, id := range statement.Predicate.Scanner.Result {
+		vulNode := assembler.VulnerabilityNode{}
```
Comment (lumjjb) [inline pkg/ingestor/parser/vuln/cerify_vuln.go:107]: nit: opinion, can use struct initializers as well

File: pkg/certifier/attestation/attestation_vuln.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// limitations under the License.
+
+package attestation_vuln
+
+import (
+	"time"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+)
+
```
Comment (lumjjb) [inline pkg/certifier/attestation/attestation_vuln.go:23]: would be helpful to also maybe add a paragraph to document the behavior or the parser, and how vuln attestations and vuln nodes are created within the graph. Just a prose will do, not required to be overly formal.

**Review by lumjjb — APPROVED** [review]
LGTM

**Pull request merged**